src/unix/core.c

```diff
@@ -938,6 +951,23 @@ int uv_getrusage(uv_rusage_t* rusage) {
   rusage->ru_nivcsw = usage.ru_nivcsw;
 #endif
 
+#if defined(__HAIKU__)
+  rusage->ru_maxrss = 0;
+  rusage->ru_ixrss = 0;
+  rusage->ru_idrss = 0;
+  rusage->ru_isrss = 0;
+  rusage->ru_minflt = 0;
+  rusage->ru_majflt = 0;
+  rusage->ru_nswap = 0;
+  rusage->ru_inblock = 0;
+  rusage->ru_oublock = 0;
+  rusage->ru_msgsnd = 0;
+  rusage->ru_msgrcv = 0;
+  rusage->ru_nsignals = 0;
+  rusage->ru_nvcsw = 0;
+  rusage->ru_nivcsw = 0;
+#endif
```

Comment:
Looking at this again, it might be better to `memset(rusage, 0, sizeof(*rusage))`, that would simplify the code and fix an AIX-only bug where we leave garbage in the unsupported fields.